### ISSUE TITLE
FODS support

### ISSUE THREAD

**Issue by paddor** [gorn/rspreadsheet#39]
Right now, I get an error when I try to open a FODS document:

```
[1] pry(main)> require 'rspreadsheet'
=> true
[2] pry(main)> doc = Rspreadsheet.open('./conf/model.fods')
Zip::Error: Zip end of central directory signature not found
from /home/p/.gem/ruby/2.6.2/gems/rubyzip-1.2.2/lib/zip/central_directory.rb:143:in `get_e_o_c_d'
```

As far as I know, it's just an uncompressed version of ODS.

**Comment by gorn** [comment]
I have never worked with FODS, but if you fill in the issue with expected usecase and some examples I can have a look at it. If you are able to file failing test, than it is even better.

**Comment by paddor** [comment]
Use case: I don’t wanna check binary files into my Git repository. ODS files are compressed and thus binary. Git can’t work out diffs of binary files.

In LibreOffice under “Save as...” I choose the FODS (Flat ODS) file format, which is an uncompressed XML file. Unfortunately this project assumes all files to be compressed and thus always uses `Zip::File` IIRC. 

Suggestion: When opening a file, check BOM of file or simply the extension in the filename provided. If it’s not compressed, don’t run it through `Zip::File`.

**Comment by gorn** [comment]
I just briefly looked at FODS file created as you describe. It seems to me in quite different format that usual ods files. Usual OSD is not only zipped, but also contains several separate files, so handeling of for example pictures is realtively easy (they are just stored in zip and referenced). We would need to find more documentation of FODS and its differencies / simmilarities to ODS before it can be implemented. Any help with this is appreciated.

**Comment by gorn** [comment]
@paddor after some investigation I think I might be able to add the support, however it would need to be tested in practical usage. Without that I do not think it is good to include it in production verions. Are you willing to do some testing, as you seem to work with fods? I personally do not use them at all.

**Comment by paddor** [comment]
Thanks, that’s great news! Yes, I’d be willing to do some testing and give feedback. To be honest though, that particular application is not reading the FODS file yet, so there’s no need to hurry. But I’d still fairly easily be able to perform some testing, whenever this feature is ready.

**Comment by gorn** [comment]
Ok, I have implemented basic functionality for fods files and released as version 0.5. It needs to be tested however, there might easily be some missing parts. Will you have a look at it @paddor ?

I know that it can not covert one format to the other and does not support images yet. Let me know.

**Comment by gorn** [comment]
I am closing this for now, but I would still be interested if @paddor or anyone else can test new functions. It works for me at the moment.
